lens-cube/src/main/java/org/apache/lens/cube/parse/StorageCandidate.java

```diff
@@ -557,4 +656,22 @@ public String toString() {
   public void addValidUpdatePeriod(UpdatePeriod updatePeriod) {
     this.validUpdatePeriods.add(updatePeriod);
   }
+
+  public void updateFromString(CubeQueryContext query, Set<Dimension> queryDims,
```

Comment:
can we move this method and related methods to CandidateUtil ?
We can update the StorageCandidate#getQueryAst#setFromString from CandidateUtil 

The objective is to have only non util like public methods in StorageCandidate and keep it clean